test/suite/Amqp/Rpc/AmqpRpcServerTest.php

```diff
@@ -469,6 +471,169 @@ function () {
+                '<response-queue>'
+            )
+        );
+
+        $context = null;
+
+        Phake::verify($this->logger)->debug(
+            'rpc.server {queue} #{id} REQUEST {procedure}({arguments})',
+            Phake::capture($context)
+        );
+
+        Phake::verify($this->logger)->debug(
+            'rpc.server {queue} #{id} RESPONSE {procedure}({arguments}) -> {code} {time} {value}',
+            Phake::capture($context)
+        );
+
+        Phake::verify($this->logger)->log(
+            LogLevel::ERROR,
+            'rpc.server {queue} #{id} {code} {procedure} {time}',
+            Phake::capture($context)
+        );
+
+        $this->assertEquals(
+            [
+                'id'        => 456,
+                'queue'     => '<response-queue>',
+                'procedure' => 'procedure-name',
+                'arguments' => '1, 2, 3',
+                'code'      => ResponseCode::EXCEPTION(),
+                'value'     => '"Internal server error."',
+                'time'      => '1.230000 ms',
+                'exception' => new Exception(
+                    'The procedure imploded spectacularly!'
+                )
+            ],
+            $context
+        );
+
+        $this->assertEquals(
+            new AMQPMessage(
+                '[' . ResponseCode::EXCEPTION . ',"Internal server error."]',
+                [
+                    'correlation_id' => 456,
+                ]
+            ),
+            $responseMessage
+        );
+    }
+
+    public function testReceiveRequestWithCustomErrorHandlerThatThrows()
+    {
+        Phake::when($this->channelDispatcher)
+            ->wait($this->channel)
+            ->thenGetReturnByLambda(
+                function () {
+                    $handler = null;
+
+                    Phake::verify($this->channel)->basic_consume(
+                        '<request-queue-procedure-name>',
+                        '',    // consumer tag
+                        false, // no local
+                        false, // no ack
+                        false, // exclusive
+                        false, // no wait
+                        Phake::capture($handler)
+                    );
+
+                    $requestMessage = new AMQPMessage(
+                        '["procedure-name",[1,2,3]]',
+                        [
+                            'reply_to' => '<response-queue>',
+                        ]
+                    );
+                    $requestMessage->delivery_info['delivery_tag'] = '<delivery-tag>';
+
+                    $handler($requestMessage);
+                }
+            )
+            ->thenReturn(null);
+
+        $someException = new Exception('Error handler throws.');
+        $exceptioned = false;
+
+        $this->server = Phake::partialMock(
+            AmqpRpcServer::class,
+            $this->logger,
+            $this->channel,
+            function (Throwable $e) { throw new Exception('Error handler throws.'); },
```

Comment:
I think this is the only line that needs to change in regards to testing for `Throwable` rather than `Exception`, just throw an `Error` to ensure it works with any `Throwable`.